From ecb10eb6b59d46051f74cdd07e694e459a84e266 Mon Sep 17 00:00:00 2001
From: Adeel Arshad <adeel_arshad@mentor.com>
Date: Wed, 19 Oct 2016 16:05:08 +0500
Subject: [PATCH] fix CXL_env update code position

In CodeXL/Components/GpuProfiling/Build/SConstruct
the position of code that was added to update the scons
consturction environment, was incorrect.

Signed-off-by: Adeel Arshad <adeel_arshad@mentor.com>
---
 CodeXL/Components/GpuProfiling/Build/SConstruct | 36 ++++++++++++-------------
 1 file changed, 18 insertions(+), 18 deletions(-)

diff --git a/CodeXL/Components/GpuProfiling/Build/SConstruct b/CodeXL/Components/GpuProfiling/Build/SConstruct
index 94f5b6a..a2e648f 100755
--- a/CodeXL/Components/GpuProfiling/Build/SConstruct
+++ b/CodeXL/Components/GpuProfiling/Build/SConstruct
@@ -13,24 +13,6 @@ from CXL_init import *
 #
 CXL_vars = Variables(None)
 
-# update CXL_env as per the external environment
-# Set defaults from environment.  Note that scons doesn't cope well
-# with multi-word CPPFLAGS/LDFLAGS/SHLINKFLAGS values; you'll have to
-# explicitly quote them or (better yet) use the "=" form of GNU option
-# settings.
-
-for key, value in os.environ.items():
-	if key in ("CFLAGS", "CCFLAGS", "LINKFLAGS"):
-		CXL_env.Replace(**{key: Split(os.getenv(key))})
-	elif key in ("LDFLAGS", "SHLINKFLAGS"):
-		pass
-	else:
-		CXL_env.Replace(**{key: os.getenv(key)})
-
-for flag in ["LDFLAGS", "SHLINKFLAGS"]:
-	if os.environ.has_key(flag):
-		CXL_env.MergeFlags({flag : [os.getenv(flag)]})
-
 # Initial CXL_vars scons construction variables
 initCXLVars(CXL_vars)
 
@@ -53,6 +35,24 @@ CXL_env = Environment(
 	variables = CXL_vars, 
 	ENV = os.environ)
 
+# update CXL_env as per the external environment
+# Set defaults from environment.  Note that scons doesn't cope well
+# with multi-word CPPFLAGS/LDFLAGS/SHLINKFLAGS values; you'll have to
+# explicitly quote them or (better yet) use the "=" form of GNU option
+# settings.
+
+for key, value in os.environ.items():
+        if key in ("CFLAGS", "CCFLAGS", "LINKFLAGS"):
+                CXL_env.Replace(**{key: Split(os.getenv(key))})
+        elif key in ("LDFLAGS", "SHLINKFLAGS"):
+                pass
+        else:
+                CXL_env.Replace(**{key: os.getenv(key)})
+
+for flag in ["LDFLAGS", "SHLINKFLAGS"]:
+        if os.environ.has_key(flag):
+                CXL_env.MergeFlags({flag : [os.getenv(flag)]})
+
 # CXL build initialization
 initCXLBuild (CXL_env)
 
-- 
1.9.1

